src/XMLimport.cpp

```diff
@@ -149,7 +152,37 @@ bool XMLimport::importPackage(QIODevice* device, QString packName, int moduleFla
         readNext();
 
         if (isStartElement()) {
-            if (name() == "MudletPackage") {
+            if (name() == QStringLiteral("MudletPackage")) {
+                QString versionString;
+                if (attributes().hasAttribute(QStringLiteral("version"))) {
+                    versionString = attributes().value(QStringLiteral("version")).toString();
+                    if (!versionString.isEmpty()) {
+                        bool isOk = false;
+                        float versionNumber = versionString.toFloat(&isOk);
+                        if (isOk) {
+                            mVersionMajor = qFloor(versionNumber);
+                            mVersionMinor = qRound(1000.0 * versionNumber) - (1000 * mVersionMajor);
+                        }
+                        if (pVersionString) {
+                            *pVersionString = versionString;
+                        }
+                    }
+                }
+
+                if (mVersionMajor > 1
+                    /*||(mVersionMajor==1&&mVersionMinor)*/) {
+                    // Minor check is not currently relevant, just abort on 2.000f or more
+
+                    QString moanMsg = tr("[ ALERT ] - Sorry, the file being read is marked as being version %1 which is\n"
```

Comment:
It turns out that `XMLimport::importPackage(...)` although having a `QIODevice *` as it's first argument is always used with a `QFile *` as that argument.  Given that QFile inherits from `QIODevice` it seems reasonable to modify the first argument to __be__ a `QFile *` so that it is possible to use a `QFile::fileName()` to get the name of the file without having to mess with `qobject_cast<QFile *>` or `dynamic_cast<QFile *>`.